src/engine/websites/decorators/MadTheme.ts

```diff
@@ -0,0 +1,80 @@
+
+const scriptImageLinks = `
+    new Promise(resolve => {
+        const images = window.chapImages.split(',');
+        resolve(images.map(image => window.mainServer + image));
+    });
+`;
+
+/* eslint-disable-next-line @typescript-eslint/no-explicit-any */ //=> A mixin class must have a constructor with a single rest parameter of type 'any[]'
+export type Constructor = new (...args: any[]) => DecoratableMangaScraper;
+
+const queryChapterListBloat = '.chapter-update';
+const queryChapterListLinks = 'ul.chapter-list li a';
+const DefaultInfoExtractor = Common.AnchorInfoExtractor(false, queryChapterListBloat);
+
+/*************************************************
+ ******** Chapter List Extraction Methods ********
+ *************************************************/
+
+async function FetchChaptersCSS(this: MangaScraper, manga: Manga, request: FetchRequest, query = queryChapterListLinks, extract = DefaultInfoExtractor): Promise<Chapter[]> {
+    const data = await FetchCSS<HTMLAnchorElement>(request, query);
+    return data.map(element => {
+        const { id, title } = extract.call(this, element);
+        return new Chapter(this, manga, id, title);
+    });
+}
+
+/**
+ * A class decorator that adds the ability to extract all chapters for a given manga from this website using the given CSS {@link query}.
+ * This method utilizes the HTML pages provided by the **MadTheme Admin AJAX endpoint** to extract the chapters.
+ * @param query - A CSS query to locate the elements from which the chapter identifier and title shall be extracted
+ * @param extract - The extractor to use
+ */
+export function ChaptersSinglePageAJAX(query = queryChapterListLinks, extract = DefaultInfoExtractor) {
+    return function DecorateClass<T extends Common.Constructor>(ctor: T): T {
+        return class extends ctor {
+            public async FetchChapters(this: MangaScraper, manga: Manga): Promise<Chapter[]> {
+                return FetchChaptersSinglePageAJAX.call(this, manga, query, extract);
+            }
+        };
+    };
+}
+
+/**
+ * An extension method for extracting all chapters for the given {@link manga} using the given CSS {@link query}.
+ * This method utilizes the HTML pages provided by the **MadTheme Admin AJAX endpoint** to extract the chapters.
+ * @param this - A reference to the {@link MangaScraper} instance which will be used as context for this method
+ * @param manga - A reference to the {@link Manga} which shall be assigned as parent for the extracted chapters
+ * @param query - A CSS query to locate the elements from which the chapter identifier and title shall be extracted
+ * @param extract - The extractor to use
+ */
+export async function FetchChaptersSinglePageAJAX(this: MangaScraper, manga: Manga, query = queryChapterListLinks, extract = DefaultInfoExtractor): Promise<Chapter[]> {
+    const id = manga.Identifier.split('/').pop();
+    const uri = new URL('/api/manga/' +id+ '/chapters?source=detail', this.URI);

```

Comment:
Personally i prefer string interpolation when inserting variable values in the middle of a string literal
```javascript
`/api/manga/${id}/chapters`
```